src/Context/products-context.js

```diff
@@ -0,0 +1,348 @@
+import { useReducer, useContext, createContext } from "react";
+
+const ProductsContext = createContext(null);
+
+const reducer = (state, action) => {
+  const encodedToken = localStorage.getItem("ENCODED_TOKEN_2");
+
```

Comment:
its a side-effect, don't do anything inside a reducer which is a side-effect, it has to be pure